hyp/base.py

```diff
@@ -51,51 +62,46 @@ def links(self, links, linked):
     def build_root_meta(self, meta):
         return meta
 
-    def build_root_links(self, links):
-        rv = {}
-
+    def build_root_links(self, collector, links):
+        # Use the collector to build the links structure
         for key in links:
-            link = self.LINKS[key]
-            association = "%s.%s" % (self.TYPE, key)
-
-            rv[association] = {'type': link['responder'].TYPE}
-            if 'href' in link:
-                rv[association]['href'] = link['href']
-
-        return rv
-
-    def build_root_linked(self, linked):
-        rv = {}
+            collector.use_link(self, key)
```

Comment:
Yeah! I agree, perhaps `append` isn't the best. But perhaps we could go with `append_link` as you're appending it to the structure that will be produced later on?